### ISSUE TITLE
Deck.gl crashing on android chrome 89 due to  GLSL compilation errors

### ISSUE THREAD

**Issue by gpng** [visgl/deck.gl#5557]
#### Description
Deck.gl is crashing on our website on android chrome, with below errors in the logs. This is also happening in https://deck.gl 

#### Expected Behavior
Deck.gl should run without errors

#### Repro Steps
Using android chrome visit https://deck.gl

#### Environment
 - Framework Version: deck.gl 8.0.9 and 8.4.9
 - Browser Version: Chrome 89.0.4389.72
 - OS: Android 10

#### Logs
```
deck: error during initialization of ArcLayer({id: 'hover-flight-arc'}) Error: GLSL compilation errors in vertex shader arc-layer-vertex-shader-2
    at VertexShader._compile (shader.js:119)
    at VertexShader.initialize (shader.js:67)
    at new Shader (shader.js:52)
    at VertexShader._createSuperInternal (shader.js:7)
    at new VertexShader (shader.js:153)
    at Program.initialize (program.js:72)
    at new Program (program.js:52)
    at ProgramManager.get (program-manager.js:151)
    at Model._checkProgram (model.js:417)
    at Model.initialize (model.js:72)
    at new Model (model.js:31)
    at ArcLayer._getModel (arc-layer.js:201)
    at ArcLayer.updateState (arc-layer.js:165)
    at ArcLayer._updateState (layer.js:786)
    at ArcLayer._initialize (layer.js:756)
    at LayerManager._initializeLayer (layer-manager.js:342)
    at LayerManager._updateSublayersRecursively (layer-manager.js:304)
    at LayerManager._updateLayers (layer-manager.js:261)
    at LayerManager.setLayers (layer-manager.js:200)
    at LayerManager.setProps (layer-manager.js:165)
    at Deck.setProps (deck.js:245)
    at Deck._setGLContext (deck.js:647)
    at Deck._onRendererInitialized (deck.js:680)
    at AnimationLoop.onInitialize (animation-loop.js:291)
    at animation-loop.js:155
```

![deckgl_crash](https://user-images.githubusercontent.com/19505419/109964360-8a74b400-7d28-11eb-8248-f134e255ec1c.gif)

**Comment by ibgreen** [comment]
Normally the actual shader compilation error log is shown, unfortunately I don't see those errors here.

Is the error reproducible in the Android emulator on a Mac? That would make it an order of magnitude easier for us to debug. Most devs likely do not have Android devices available.

Can you break in luma.gl Shader._compile and see if you get any errors back from the compiler?

**Comment by gpng** [comment]
Ah unfortunately I don't have an android emulator on a mac at this time.

The only error message being printed from _compile is `shader.js:111 luma.gl: GLSL compilation errors in vertex shader arc-layer-vertex-shader-2`

I see that it's supposed to concat the errors after the shader name, but it looks like there isn't any

**Comment by ibgreen** [comment]
> Ah unfortunately I don't have an android emulator on a mac at this time.

Can you replicate it on an Android emulator (regardless of OS)?

If it is a device specific issue it will not be easy for us to fix it.

**Comment by gpng** [comment]
Ah okay i've managed to replicate it on an emulator. I can't find any emulators that can update the main chrome app to the latest version  89.0.4389.72, but using the official chrome beta app (also at 89.0.4389.72) manages to replicate this issue

**Comment by gpng** [comment]
Inspecting the errors on the emulated Pixel 31 API 30, it seems it gives a different error, hopefully it helps

on deck.gl
```
deck: error during initialization of TripsLayer({id: 'trips'}) Error: WebGL instanced rendering not supported
    at r (app-775e1336109a8d94acb0.js:2)
    at Object.vertexAttribDivisor (app-775e1336109a8d94acb0.js:2)
    at app-775e1336109a8d94acb0.js:2
    at n.value (app-775e1336109a8d94acb0.js:2)
    at n.value (app-775e1336109a8d94acb0.js:2)
    at e.value (app-775e1336109a8d94acb0.js:2)
    at e.value (app-775e1336109a8d94acb0.js:2)
    at app-775e1336109a8d94acb0.js:2
    at n.value (app-775e1336109a8d94acb0.js:2)
    at e.value (app-775e1336109a8d94acb0.js:2)
```

**Comment by ibgreen** [comment]
- If instanced rendering is not supported by WebGL on a platform, most deck.gl layers will be not be able to run. 
- Unless there is a software update there is not much that we can do. We either need WebGL2 or WebGL1 with instanced rendering extension.
- You may want to Google a bit about what WebGL versions are supported on different Android releases and devices.

I found this table, might give some insight? https://gist.github.com/guimaraf/93137e73ca5d395bd806c51a6a3c80b5

**Comment by Pessimistress** [comment]
I have a Pixel 4a and was unable to reproduce the issue (Android 11 and Chrome 88). If this is only in Chrome 89 there might be an issue with this release of Chrome? Are you able to run https://threejs.org/examples/webgl_interactive_cubes.html ?

Are you testing in an emulator or an actual device?

**Comment by gpng** [comment]
I would think it is a chrome 89 issue as I have been able to use my site with deckgl with no issues until the latest update. 

https://threejs.org/example/webgl_interactive_cubes.html runs fine on my device.

I'm primarily testing on an actual device, but was using various emulators to see if the issue could be reproduced. The issue was also first brought up to us by a user also on Android chrome 89

**Comment by nevets963** [comment]
I also have the same issue using an old version of Deckgl (7.0.10).

Although this is from the minified version, I believe it's the same compilation issue mentioned:
Uncaught Error: GLSL compilation errors in vertex shader solid-polygon-layer-vertex-shader-1
    at t.value (deck.gl-7.0.10-dist.min.js:1)
    at t.value (deck.gl-7.0.10-dist.min.js:1)
    at t (deck.gl-7.0.10-dist.min.js:1)
    at new t (deck.gl-7.0.10-dist.min.js:1)
    at e.value (deck.gl-7.0.10-dist.min.js:1)
    at e.value (deck.gl-7.0.10-dist.min.js:1)
    at e.value (deck.gl-7.0.10-dist.min.js:1)
    at t.value (deck.gl-7.0.10-dist.min.js:1)
    at t.value (deck.gl-7.0.10-dist.min.js:1)
    at t.value (deck.gl-7.0.10-dist.min.js:1)

@gpng @ibgreen @Pessimistress Has anyone submitted a bug report already with Google? Just want to check before I submit one

**Comment by nevets963** [comment]
I have submitted a ticket. If anyone has anything to add, this can be done here: https://bugs.chromium.org/p/chromium/issues/detail?id=1184837

**Comment by ibgreen** [comment]
So initial feedback from Google is that the issues occur because Chrome GPU process runs out of memory and issue has been escalation. 

Unfortunately, deck.gl maintainers do not regularly test on Android devices or optimize heavily for mobile. We do have a browser stack account at our disposal that enables some of the necessary testing but we do not have a good automated test setup. 

If there are serious mobile users of deck.gl out there who depend on deck.gl performance on mobile and would consider supporting deck.gl in this area (help test new deck.gl releases and test new Android/Chrome releases) that would be very valuable. 

Note that deck.gl has a full contributor program and is under open governance through the Linux Foundation.

**Comment by nekitk** [comment]
We've ran into the same issue on multiple Android devices: Google Pixel 4A, Samsung Galaxy S20.

Now chromium issue is marked as fixed and it was shipped in the latest Chrome release for Android.

Tested on our devices - issue went away after update.

**Comment by nevets963** [comment]
@nekitk Great to hear! Out of interest were you able to simply update Chrome through Google Play immediately? On my Android device, I don't yet see the update available

For reference and other folks, here is the update release page: https://chromereleases.googleblog.com/2021/03/chrome-for-android-update_8.html

**Comment by nekitk** [comment]
> @nekitk Great to hear! Out of interest were you able to simply update Chrome through Google Play immediately? On my Android device, I don't yet see the update available

Yes, update 89.0.4389.86 was available at Google Play. Maybe it differs bases on region/device. Interesting thing is that other Pixel device on our team is still on Chrome 88 and doesn't have Chrome update available.

**Comment by chris-aeviator** [comment]
We are running into this while testing a deck.gl based MVP. A group of users reported their inability to access the product at all (due to a WSoD https://en.wikipedia.org/wiki/Screen_of_death#White ) on android devices in a recent feedback round.

**Comment by gpng** [comment]
Confirming that issue seems to be fixed on Chrome  89.0.4389.86 update received through Google play.
